portality/api/v1/discovery.py

```diff
@@ -121,13 +121,24 @@ def _sanitise(cls, q, page, page_size, sort, search_subs, sort_subs):
         if page < 1:
             page = 1
 
-        if page_size > app.config.get("DISCOVERY_MAX_PAGE_SIZE", 100):
+        if bulk :
+            max_pag_size = app.config.get("DISCOVERY_BULK_PAGE_SIZE", 1000)
+        else:
+            max_pag_size = app.config.get("DISCOVERY_MAX_PAGE_SIZE", 1000)
+        if page_size > max_pag_size:
             page_size = app.config.get("DISCOVERY_MAX_PAGE_SIZE", 100)
         elif page_size < 1:
             page_size = 10
-
         # calculate the position of the from cursor in the document set
         fro = (page - 1) * page_size
+        # If fro is greater than the max allowed, throw error
+        # using bulk to provide an override when needed
+        max_records = app.config.get("DISCOVERY_MAX_RECORDS_SIZE", 1000)
+        if not bulk and fro >= max_records:
+            message = """Looks like you want to view results beyond {x}. \
+If you would like to see more results, you can download all of our results \
+from link""".format(x=max_records)
```

Comment:
This is a placeholder for the full data dump URL, but this should be either be meaningful without that or we shouldn't release this until the dump is ready.